src/h3/mod.rs

```diff
@@ -689,11 +689,24 @@ impl Connection {
             return Err(Error::WrongStream);
         }
 
+        let overhead = octets::varint_len(frame::DATA_FRAME_TYPE_ID) +
+            octets::varint_len(body.len() as u64);
+
+        let stream_cap = conn.stream_capacity(stream_id)?;
+
+        // Check if there is enough capacity to send the frame header
+        if stream_cap < overhead {
+            return Err(Error::Done);
+        }
+
+        // Cap the frame payload length to the stream's capacity.
+        let body_len = std::cmp::min(body.len(), stream_cap - overhead);
```

Comment:
If stream_cap == overhead, it could result in sending 0-payload-length DATA frames. They are technically fine, but I think it adds work for library and the calling app to deal with the fact that 0-non-header bytes were written.

Reply:
Yeah, good point. I can't imagine a use-case for this either, so we can just forbid it by changing the check above to `if stream_cap <= overhead`.

Reply:
Fixed.